### DISCUSSION TITLE
How do I currently disable and hide fields in the checkout?

### DISCUSSION THREAD

**Discussion by KPABOOKING** [woocommerce/woocommerce#54063] [Checkout Flow]
I use the checkout block and have tried many codes, until recently everything worked, after the update - making optional and hiding fields is just not possible.
I tried this code as well:

```php
add_filter('woocommerce_get_country_locale', function( $locale ) {
	foreach ( $locale as $key => $value ) {
		$locale[ $key ]['address_1'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['postcode'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['city'] = [
			'required' => false,
			'hidden'   => true,
		];

		$locale[ $key ]['state'] = [
			'required' => false,
			'hidden'   => true,
		];
	}

	return $locale;
});

```

This code is presented to us as official - but it doesn't work.
https://developer.woocommerce.com/docs/cart-and-checkout-removing-checkout-fields/
I only sell to one country (Russia).
It didn't help

**Comment by nielslange** [comment]
Hello @KPABOOKING 👋

Thanks for reaching out. On my end, both your code snippet and the following code snippet are leading to the expected result:

```php
add_filter('woocommerce_get_country_locale', function( $locale ) {

	$locale['RU']['address_1']['required'] = false;
	$locale['RU']['address_1']['hidden'] = true;

	$locale['RU']['postcode']['required'] = false;
	$locale['RU']['postcode']['hidden'] = true;

	$locale['RU']['city']['required'] = false;
	$locale['RU']['city']['hidden'] = true;

	$locale['RU']['state']['required'] = false;
	$locale['RU']['state']['hidden'] = true;

	return $locale;
});
```

In both cases, I see the following result:

<img width="721" alt="Screenshot 2025-01-07 at 16 41 57" src="https://github.com/user-attachments/assets/6be810f5-974e-4de7-b30f-8dffd4d0229b" />

You mentioned that after the update, the code snippet stopped working. Are you referring to updating WooCommerce itself or any other plugin? As the code snippets work on my end, I suggest executing [a plugins and theme conflict test](https://woocommerce.com/document/how-to-test-for-conflicts/) to check if your theme or one of your activated plugins is causing the issue that you are currently seeing.
